src/signed_data.rs

```diff
@@ -352,84 +352,124 @@ pub static RSA_PSS_2048_8192_SHA384_LEGACY_KEY: SignatureAlgorithm = SignatureAl
 /// Requires the `alloc` feature.
 #[cfg(feature = "alloc")]
 pub static RSA_PSS_2048_8192_SHA512_LEGACY_KEY: SignatureAlgorithm = SignatureAlgorithm {
-    public_key_alg_id: RSA_ENCRYPTION,
-    signature_alg_id: RSA_PSS_SHA512,
+    public_key_alg_id: alg_id::RSA_ENCRYPTION,
+    signature_alg_id: alg_id::RSA_PSS_SHA512,
     verification_alg: &signature::RSA_PSS_2048_8192_SHA512,
 };
 
 /// ED25519 signatures according to RFC 8410
 pub static ED25519: SignatureAlgorithm = SignatureAlgorithm {
-    public_key_alg_id: ED_25519,
-    signature_alg_id: ED_25519,
+    public_key_alg_id: alg_id::ED_25519,
+    signature_alg_id: alg_id::ED_25519,
     verification_alg: &signature::ED25519,
 };
 
-struct AlgorithmIdentifier {
-    asn1_id_value: untrusted::Input<'static>,
-}
-
-impl AlgorithmIdentifier {
-    fn matches_algorithm_id_value(&self, encoded: untrusted::Input) -> bool {
-        encoded == self.asn1_id_value
+/// Encodings of the PKIX AlgorithmIdentifier type:
+///
+/// ```text
+/// AlgorithmIdentifier  ::=  SEQUENCE  {
+///     algorithm               OBJECT IDENTIFIER,
+///     parameters              ANY DEFINED BY algorithm OPTIONAL  }
+///                                -- contains a value of the type
+///                                -- registered for use with the
+///                                -- algorithm object identifier value
+/// ```
```

Comment:
nit: maybe add a reference to where this ASN.1 definition came from? https://www.rfc-editor.org/rfc/rfc5280#section-4.1.1.2 ?